cyder/base/fields.py

```diff
@@ -51,12 +51,15 @@ def to_python(self, value):
         value = super(MacAddrField, self).to_python(value)
 
         if value:
-            value = value.lower().replace(':', '').replace('-', '')
-            value = reduce(lambda x,y: x + ':' + y,
-                           (value[i:i+2] for i in xrange(0, 12, 2)))
-        elif value == '':
-            value = None
-        return value
+            raw = value.lower().replace(':', '').replace('-', '')
+            if len(raw) != 12:
+                # We can't raise a ValidationError in to_python() (because
+                # nothing will catch it), so all we can do is make sure clean()
+                # will raise a ValidationError when it's called next.
+                return 'TOO LONG'
```

Comment:
return constants.error.mac_too_long
mac_too_long = "Error: Mac Address is too long"
Then in validate_mac
elif mac = constants.error.mac_too_long:
        raise ValidationError(mac)

We should slowly shift over to having all user visible errors/msgs in constants.